great_expectations/data_context/store/datasource_store.py

```diff
@@ -126,24 +126,27 @@ def deserialize(
     @override
     def ge_cloud_response_json_to_object_dict(
         self, response_json: CloudResponsePayloadTD  # type: ignore[override]
-    ) -> dict:
+    ) -> dict | list[dict]:
         """
         This method takes full json response from GX cloud and outputs a dict appropriate for
         deserialization into a GX object
         """
         logger.debug(f"GE Cloud Response JSON ->\n{pf(response_json, depth=3)}")
         data = response_json["data"]
         if isinstance(data, list):
-            if len(data) > 1:
-                # TODO: handle larger arrays of Datasources
-                raise TypeError(
-                    f"GX Cloud returned {len(data)} Datasources but expected 1"
-                )
-            data = data[0]
+            if len(data) == 1:
+                data = data[
+                    0
+                ]  # If a single datasource is returned in list form, extract and treat as individual
```

Comment:
This is just keeping legacy behavior but I feel a little iffy dealing with a list when we have just one element - the backend should provide a consistent structure in my opinion (maybe always a list even if one element?)